### DISCUSSION TITLE
MME support multiple TAIs?

### DISCUSSION THREAD

**Discussion by jpad-freedomfi** [magma/magma#10679] [Q&A] [answered]
Hi, 

We have a customer trying to use Magma MME/AGW in a RAN Sharing / MOCN setup. The S1 Setup Request comes with 2 TAIs, one for home network of RAN, and one for customer network. 

MME is rejecting the S1 Setup saying unknown PLMN. 

I see here MME can support 16 TAIs, making me think we can add the RAN owner PLMN/TAI here. But then there is a note with 3 !!! saying actually only one. 

https://github.com/magma/magma/blob/f6d1e28b47b55944453bd0a41f9fe91ec5ee12ae/lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_config.cpp#L114 

Anyone know if we can add > 1 TAI and if that will work? if so what is the syntax? 

@ulaskozat

**Comment by jpad-freedomfi** [comment]
Update: Looking at code, looks like it should be finding the 2nd PLMN in the SupportTAs Item. 

Looking at 
https://github.com/magma/magma/blob/66fc3fcebc73ce6c676c76eab1ee67d72568e3cb/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_ta.c#L123 

which gets called by:
https://github.com/magma/magma/blob/ec4f30d8848d0f3e15cef47025a38f9b8b501bfe/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c#L593

our S1setup request has 2 PLMN in the list. The second is matching the magma network configuration.

**Comment by jpad-freedomfi** [comment, marked as answer]
Update, we added the eNB to the orc8r/NMS as externally managed, made sure the TAC was configured to be matching, and now the S1setup works.

**Comment by ulaskozat** [comment]
@jpad-freedomfi unfortunately unless you ping me on slack, most of the discussion items go into a very noisy and crowded mailbox folder. So I missed this thread. But looks like you solved the problem? :)

**Reply by jpad-freedomfi** [reply]
Best case right! Thank you. I will ping if/when things become urgent.
